src/field_specials.c

```diff
@@ -3421,40 +3421,42 @@ bool8 sub_813B260(void)
     return FALSE;
 }
 
-void sub_813B2E4(void)
+void CreateUnusualWeatherEvent(void)
 {
     u16 randomValue = Random();
-    VarSet(VAR_0x4038, 0);
+    VarSet(VAR_UNUSUAL_WEATHER_STEP_COUNTER, 0);
 
     if (FlagGet(FLAG_DEFEATED_KYOGRE) == TRUE)
     {
-        VarSet(VAR_0x4037, (randomValue & 7) + 1);
+        VarSet(VAR_UNUSUAL_WEATHER_LOCATION, (randomValue & (UNUSUAL_WEATHER_COUNT_PER_LEGENDARY - 1)) + UNUSUAL_WEATHER_GROUDON_LOCATIONS_START);
```

Comment:
does this still match if the `&` is changed to a modulus?  (Remember to not subtract 1 when using modulus)

Reply:
Yup!